lib/rubygems/command.rb

```diff
@@ -365,6 +366,18 @@ def remove_option(name)
     end
   end
 
+  def deprecate_option(short_name: nil, long_name: nil)
```

Comment:
Can you add a option about deprecated version like this

```
deprecate_option(long_name: "--no-ri", version: 3.1)
```

We can show with above method:

```
The \"#{option}\" option has been deprecated and will be removed in future versions of Rubygems 3.1, its use is discouraged."
```